plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -467,48 +513,53 @@ private class AddFileEntryExtractor
         @Override
         public DeltaLakeTransactionLogEntry getEntry(ConnectorSession session, int pagePosition, Block... blocks)
         {
-            checkState(blocks.length == getRequiredBlocks(), "Unexpected amount of blocks: %s", blocks.length);
-            Block block = blocks[0];
-            log.debug("Building add entry from %s pagePosition %d", block, pagePosition);
-            if (block.isNull(pagePosition)) {
+            checkState(blocks.length == getRequiredChannels(), "Unexpected amount of blocks: %s", blocks.length);
+            Block addBlock = blocks[0];
+            Block addPartitionValuesBlock = blocks[1];
+            log.debug("Building add entry from %s pagePosition %d", addBlock, pagePosition);
+            if (addBlock.isNull(pagePosition) || addPartitionValuesBlock.isNull(pagePosition)) {
```

Comment:
The `isNull` check for `addPartitionValuesBlock` is being added because there are now 2 blocks (instead of initially 1) from which we build up the add entry and they need to be consistent. 
Changing though slightly the logic. Thank you for raising this.